src/librustc/hir/map/mod.rs

```diff
@@ -1032,6 +1032,7 @@ impl<'hir> Map<'hir> {
             // unit/tuple structs take the attributes straight from
             // the struct definition.
             Some(Node::StructCtor(_)) => return self.attrs(self.get_parent(id)),
+            Some(Node::Crate) => Some(&self.forest.krate.attrs[..]),
```

Comment:
This actually has a separate API from regular attribute access, in several places, it would be nice if we could get rid of the other APIs.